std/math/exponential.d

```diff
@@ -69,9 +69,6 @@ if (isFloatingPoint!(F) && isIntegral!(G))
     import std.math.traits : isNaN;
     import std.traits : Unsigned;
 
-    // NaN ^^ 0 is an exception defined by IEEE (yields 1 instead of NaN)
-    if (isNaN(x)) return n ? x : 1.0;
-
```

Comment:
`n == 0` already returns and used to return `1.0`. A `NaN` should be already propagated in the other cases too.

Reply:
Is there a test that covers this execution path?

Reply:
I don't know; no test was added in #7823 in the first place though, and nobody complained back then.